xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +80,38 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights and updates MSet.
+     *  Dereferencing the Iterator should return a double.
+     *  Iterator should dereference weights in the original order of documents.
+     *  Weights are assigned as per the original order of documents.
+     *
+     *  @param begin	Begin iterator.
+     *  @param end	End iterator.
+     *
+     *  @exception Xapian::InvalidArgument is thrown if the total number of
+     *  elements in the input doesn't match the total number of documents in MSet.
+     *
+     */
+    template <typename Iterator>
+    void replace_weights(Iterator first, Iterator last)
+    {
+	if (last - first != size()) {
+	    throw Xapian::InvalidArgumentError("Number of weights assigned doesn't match the number of items.");
+	}
+	Xapian::doccount i = 0;
+	while (first != last) {
+	    set_item_weight(i, *first);
+	    ++i;
+	    ++first;
+	}
+    }
+
+    /**
+     * Sorts the list of documents in MSet according to their weights.
+     * Use after calling MSet::replace_weights.
```

Comment:
Also the brief description should be on the first line (i.e. `/** Sorts the list of documents in the MSet according to their weights.`)